Improve map components

tnagorra opened the pull request:
## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

frozenhelium commented on app/components/DefaultMapContainer/index.tsx:
```diff
@@ -14,14 +17,24 @@ function DefaultMapContainer(props: Props) {
+                className={_cs(
+                    styles.map,
+                    compact && styles.compact,
+                    className,
```
Lets make it `mapContainerClassname`

tnagorra committed:
fix(map): change default configuration of base map

- zoom is 1
- tileSize is 512

tnagorra committed:
fix(map): update tilesize and canvas size

tnagorra committed:
fix(map): "zoom level" should be one less of "tile z"

tnagorra committed:
fix(map): add debouncing before mounting map layers

- add option to zoom to a feature ensuring it falls in a single tile
- refactor CompletenessScenarioPreview to use GeoJsonPreview instead of
  GeoJsonMapSource
- remove padding by default
- add options to override zoom level and bounds on maps
- pass default bounds for find, compare, completenss and validate
  project
- show default bounds for validate project

tnagorra committed:
chore(submodule): update backend
